src/ott/tools/sliced.py

```diff
@@ -0,0 +1,75 @@
+# Copyright OTT-JAX
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Callable, Optional, Tuple
+
+import jax
+import jax.numpy as jnp
+
+from ott.geometry import costs, pointcloud
+from ott.solvers import linear
+
+__all__ = ["sliced_w"]
+
+Randfunc_t = Callable[[jax.Array, (int, int)], jnp.ndarray]
+
+
+def sliced_w(
+    x: jnp.ndarray,
+    y: jnp.ndarray,
+    a: jnp.ndarray,
+    b: jnp.ndarray,
+    cost_fn: Optional[costs.CostFn] = None,
+    proj_matrix: Optional[jnp.ndarray] = None,
```

Comment:
I'd just pass `proj_matrix: Optional[jnp.ndarray] = None` + define a function in this file, e.g.:
```python
def random_direction(rng: jax.Array, shape: Tuple[int, int]) -> jnp.ndarray:
    pass
# or

def random_direction(dim: int, n_proj: int = 100, rng: Optional[jax.Array] = None) -> jnp.ndarray:
    pass
``` 
and use it as
```python
if proj_matrix is None:
    n_proj = 100
    proj_matrix = random_direction(rng, (n_proj, dim))
```

Reply:
used something a bit simpler. API is still easy (but we're hiding some params that i would expect the user would prefer being exposed to, chiefly among them `n_proj`)